src/easylink/step.py

```diff
@@ -815,6 +815,27 @@ def set_configuration_state(
             parent_config, combined_implementations, input_data_config
         )
 
+    def _duplicate_template_step(self) -> Step:
+        """Makes a duplicate of the template ``Step``.
+
+        A duplicate of the :attr:`template_step`, i.e. the ``Step`` that may have
+        multiplicity, should contain copies of every attribute of said ``Step``
+        *except* for the :attr:`parent_step`. This is because :attr:`parent_step`
+        contains an actual ``Step`` instance and we want all of the duplicates to
+        have that same parent as opposed to copies of one.
+
+        Returns
+        -------
+            A duplicate of the :attr:`templated_step`.
+        """
+        # Due to the cyclic nature, self == self.template_step.parent_step
+        parent_step = self
+        self.template_step.parent_step = None
+        step_copy = copy.deepcopy(self.template_step)
+        step_copy.set_parent_step(parent_step)
+        self.template_step.set_parent_step(parent_step)
+        return step_copy
```

Comment:
Another alternative, which admittedly might do a lot more copying:

```suggestion
        step_copy = copy.deepcopy(self.template_step)
        # The naive deepcopy has also copied the `parent_step` attribute;
        # we don't want this to be pointing to a *copy* of `self`, but to the
        # original!
        step_copy.set_parent_step(self)
        return step_copy
```

The comment also possibly replaces the note in the docstring.